src/main/java/seedu/recipe/model/recipe/ingredient/Ingredient.java

```diff
@@ -1,9 +1,75 @@
 package seedu.recipe.model.recipe.ingredient;
 
+import static seedu.recipe.commons.util.AppUtil.checkArgument;
+
 /**
  * Represents an ingredient that is used in a {@code Recipe}.
  */
 public class Ingredient {
-    public static final String MESSAGE = "";
-    public static final String VALIDATION_REGEX = "\\S+(\\s+\\S+)*";
+    public static final String MESSAGE_CONSTRAINTS = "An ingredient should be made up of one or more groups of "
+            + "whitespace separated alphabetic characters. These characters may also be separated by "
+            + "singular hyphens, such as 'self-raising flour'.";
+    private static final String WORD_GROUP = "[A-Za-z]+(\\-[A-Za-z]+)?";
+    public static final String VALIDATION_REGEX = String.format("^%s(\\s+%s)*$", WORD_GROUP, WORD_GROUP);
+
+    //Data fields
+    private final String name;
+    private String commonName = "";
+
+    private Ingredient(String name) {
+        this.name = name;
```

Comment:
Would it be better to initialize `commonName` to the name of the ingredient instead of leaving it blank? Then if there are any "more common names" in a sense then we can set it using `setCommonName` like how it was intended to be.

Reply:
We use this as a state check to output the toString accordingly, so no